### PR TITLE
LITE-17948: Add bootstrap report project

### PR DESCRIPTION
**Pull request by jpdolz**

### CODE REVIEW

**Review by marcserrat — CHANGES_REQUESTED** [review]
Bootstrap shall NOT use .data folder, but rather current workdir
also we must check if exists already and control cookiecutter exception

File: connect/cli/plugins/project/commands.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+@grp_project_report.command(
+    name='bootstrap',
+    short_help='Bootstrap new report project.',
+)
+@click.option(
+    '--data-dir', '-d',
+    default=os.getcwd(),
+    type=click.Path(exists=False, file_okay=False, dir_okay=True),
+    required=True,
+    help='Directory where the report project will be deployed.',
```
Comment (marcserrat) [inline connect/cli/plugins/project/commands.py:35]: deployed => created

File: connect/cli/plugins/project/commands.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+@grp_project_report.command(
+    name='list',
+    short_help='List report projects.',
+)
+@click.option(
+    '--data-dir', '-d',
+    default=os.getcwd(),
+    type=click.Path(exists=True, file_okay=False, dir_okay=True),
+    help='Directory of report projects.',
```
Comment (marcserrat) [inline connect/cli/plugins/project/commands.py:49]: Working directory for report projects

File: connect/cli/plugins/project/helpers.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    click.secho('Bootstraping report project...\n', fg='blue')
+
+    # Avoid asking rewrite clone boilerplate project
+    cookie_dir = DEFAULT_CONFIG['cookiecutters_dir']
+    rmtree(cookie_dir)
+    try:
+        project_dir = cookiecutter(BOILERPLATE_URL, output_dir=data_dir)
+        click.secho(f'\nReport Project location: {project_dir}', fg='blue')
+    except OutputDirExistsException as error:
+        raise ClickException(str(f'\n{error}'))
```
Comment (marcserrat) [inline connect/cli/plugins/project/helpers.py:34]: Maybe lets add a more human readable error, "already exists a project directory called XXX from one of your existing projects, if you would like to use same name delete the directory first."

**Review by marcserrat — CHANGES_REQUESTED** [review]
Small improvements on texts

File: connect/cli/plugins/project/helpers.py (hunk 3 above)
Comment (jpdolz) [reply connect/cli/plugins/project/helpers.py:34]: "The directory XXXX is already created, if you would like to use that name, please delete the directory or use another location."
What do you think @marcserrat ?

File: connect/cli/plugins/project/commands.py (hunk 2 above)
Comment (jpdolz) [reply connect/cli/plugins/project/commands.py:49]: Fixed!

File: connect/cli/plugins/project/commands.py (hunk 1 above)
Comment (jpdolz) [reply connect/cli/plugins/project/commands.py:35]: Fixed!

**Review by marcserrat — APPROVED** [review]

File: connect/cli/plugins/project/helpers.py [hunk 4]
```diff
@@ -59,3 +67,63 @@ def list_projects(data_dir: str):
...
+        )
+    return data
+
+
+def _entrypoint_validations(project_dir, entrypoint, report_spec):
+    # Package validation
+    sys.path.append(os.path.join(os.getcwd(), project_dir))
+    package = entrypoint.rsplit('.', 1)[0]
+    try:
+        entrypoint_module = inspect.importlib.import_module(package)
```
Comment (ffaraoneim) [inline connect/cli/plugins/project/helpers.py:114]: You should use import_module from importlib, not through inspect.

File: connect/cli/plugins/project/helpers.py [hunk 5]
```diff
@@ -59,3 +67,63 @@ def list_projects(data_dir: str):
...
+    except Exception as error:
```
Comment (ffaraoneim) [inline connect/cli/plugins/project/helpers.py:115]: Too generic, should catch ImportError

**Review by ffaraoneim — CHANGES_REQUESTED** [review]

**Pull request closed without merging**